modules/library/render/src/main/java/org/geotools/renderer/lite/MemoryFilterOptimizer.java

```diff
@@ -111,7 +117,58 @@ public Object visit(Not filter, Object extraData) {
 
     @Override
     public Object visit(Or filter, Object extraData) {
-        return memoize(filter, extraData, super::visit);
+        return memoize(filter, extraData, this::inFilterOptimizer);
+    }
+
+    /** Checks if an Or filter can be replaced by a single IN condition. */
+    private Object inFilterOptimizer(Or filter, Object extraData) {
+        InFunction inFilter = replaceWithInFilter(filter);
+        if (inFilter == null) return super.visit(filter, extraData);
+        return ff.equals(inFilter, ff.literal(true));
+    }
+
+    private InFunction replaceWithInFilter(Or filter) {
+        List<Filter> children = filter.getChildren();
+        if (children.size() == 0
```

Comment:
This code does repeated passes over data structures as long as the or children, while a traditional loop could have done everything in a single pass, just keeping track of the first property name found, and the list of literals found in the process. 

Just saying, streams are not always an improvement, had this been in a performance critical section of code, the penalty would have been severe. But it's in a one-off, so it does not matter much, you can leave it as is.

Reply:
Indeed a single iteration over the collection is possible and faster, so I did the fix, thanks!